services/w3c/w3c-validation.service.js

```diff
@@ -0,0 +1,115 @@
+'use strict'
+const Joi = require('@hapi/joi')
+const { optionalUrl } = require('../validators')
+const {
+  documentation,
+  presetRegex,
+  getColor,
+  getMessage,
+  getSchema,
+} = require('./w3c-validation-helper')
+const { BaseJsonService } = require('..')
+
+const schema = Joi.object({
+  messages: Joi.array()
+    .required()
+    .items(
+      Joi.object({
+        type: Joi.any()
+          .allow('info', 'error', 'non-document-error')
+          .required(),
+      })
+    ),
+}).required()
+
+const queryParamSchema = Joi.object({
+  targetUrl: optionalUrl.required(),
+  preset: Joi.string()
+    .regex(presetRegex)
+    .allow(''),
+}).required()
+
+module.exports = class W3cValidation extends BaseJsonService {
+  static get category() {
+    return 'analysis'
+  }
+
+  static get route() {
+    return {
+      base: 'w3c-validation',
+      pattern: ':parser(default|html|xml|xmldtd)',
+      queryParamSchema,
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'W3C Validation',
+        namedParams: { parser: 'html' },
+        queryParams: {
+          targetUrl: 'https://about.validator.nu',
+          preset: 'HTML, SVG 1.1, MathML 3.0',
+        },
+        staticPreview: this.render({ messageTypes: {} }),
+        documentation,
+      },
+    ]
+  }
+
+  static get defaultBadgeData() {
+    return {
+      label: 'w3c',
+    }
+  }
+
+  static render({ messageTypes }) {
+    return {
+      label: 'w3c',
+      message: getMessage(messageTypes),
+      color: getColor(messageTypes),
+    }
+  }
+
+  async fetch(targetUrl, preset, parser) {
+    const qs = { doc: targetUrl, out: 'json' }
+    if (preset) qs['schema'] = getSchema(preset)
+    if (parser && parser.toLowerCase() != 'default') qs['parser'] = parser
```

Comment:
Since `parser` is a route param with a fixed set of values, it will always be one of those allowed values (and all are lowercase), so we can simplify this a bit. I've inlined the conditional below in the suggested changes for `qs`
```suggestion
```